hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/TestApplicationLimits.java

```diff
@@ -194,6 +207,8 @@ public void testAMResourceLimit() throws Exception {
     Resource clusterResource = Resource.newInstance(80 * GB, 40);
     root.updateClusterResource(clusterResource, new ResourceLimits(
         clusterResource));
+    LeafQueue queue = (LeafQueue) root.getChildQueues().stream().filter(
+        child -> child.getQueueName().equals(A)).findFirst().orElseThrow(NoSuchElementException::new);
```

Comment:
TBH, this is not mine code, it was written by @9uapaw but i will try to find out

Reply:
Its been a while since I made these changes, but as I remember this phenomenon was repeated multiple times in several tests and caused a lot of headaches. Essentially, the original LeafQueue which was spied upon is not part of the queue hierarchy, but instantiated separately in this test file only, therefore it will not be a part of the calculation steps either. IMO we should never initialize queues like this.